added let's encrypt certbot as install option

tiedtoastar opened the pull request:
Autocert bot will make it more convenient for people to setup their jellyfin.
Especially with most mobile apps requiring self signed certs to be imported into the OS itself for a connection to be established.

Commit:
# added let's encrypt certbot as install option
# modified the requirements.yml: added geerlingguy.certbot
# modified the defaults
# added certbot.yml for tasks relating to certbot
# main.yml now includes certbot task

Commit:
Build was failing
added new line in certbot.yml
self signed cert should be the default option, because autocertbot shouldn't be test as travisci doesn't have a domain to test

ghost commented:
I believe the build is failing because jellyfin_FQDN is now a string, rather than an array.
The array doesn't match: [a.example.com, b.example.com] within your self cert ansible role so the build is failing. Is there any reason why you don't use a * cert? Maybe this role should be decoupled from the self cert ansible role?

    changed: [instance] => (item=/etc/ssl/private/['a.example.com', 'b.example.com'])

    changed: [instance] => (item=/etc/ssl/csr)

    ok: [instance] => (item=/etc/ssl/private/['a.example.com', 'b.example.com'])

Commit:
adding the repo step was wrong, Debian had to be modified with {{ ansible_distribution|lower }}

Commit:
https requests were not working.

Now using the recommended settings from jellyfin
https://jellyfin.org/docs/general/administration/reverse-proxy.html
